rust/connlib/tunnel/src/client.rs

```diff
@@ -99,6 +95,56 @@ where
         Ok(())
     }
 
+    pub fn remove_resource(&mut self, id: ResourceId) {
```

Comment:
This function handle quite a lot of edge cases to make sure we clean everything properly, it's probably a good indicator that we want to refactor how we spread the links between resources ids and the different parts of the state but it works for now.